New module to query Lowell Observatory astorbDB database

hhsieh00 opened the pull request:
This is a new module for querying the Lowell Observatory astorbDB database (see issue #3129), specifically using REST APIs used to run their AstInfo tool (https://asteroid.lowell.edu/astinfo/).  The AstInfo class in this module specifically retrieves designation, orbital element, orbit, albedo, color, taxonomy, lightcurve, dynamical family, and escape route data for asteroids catalogued in the Lowell astorbDB database.

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
draft astroquery code for Lowell astorbdb AstInfo functionality

query code only for now; tests not written yet

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
Fixing some codestyle issues in core.py

Fixing some codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
fixing more codestyle issues in core.py

fixing more codestyle issues in core.py

hhsieh00 committed:
more codestyle issue fixes

hhsieh00 committed:
more codestyle issue fixes

hhsieh00 committed:
more codestyle issue fixes

hhsieh00 committed:
more codestyle issue fixes

mkelley commented on astroquery/astorbdb/__init__.py:
```diff
@@ -0,0 +1,39 @@
+
+    timeout = _config.ConfigItem(
+        30,
+        'Time limit for connecting to template_module server.')
```
Fix "template_module"

hhsieh00 replied:
fixed

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+    _uri = None
+
+    def designations_async(self, object_name, *,
+                           get_raw_response=False,
```
get_raw_response keyword argument should not be used (the async method always returns a raw response)

mkelley replied:
Similarly, remove the related if statement, and edit the other methods.

hhsieh00 replied:
done

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+
+    def designations_async(self, object_name, *,
+                           get_raw_response=False,
+                           get_uri=False,
```
I don't think this is needed either.  Maybe you were following the jplhorizons module? (which is ancient frankencode)

mkelley replied:
Similarly, remove the related if statement, and edit the other methods.

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+
+        return response
+
+    def dynamicalfamily_async(self, object_name, *,
```
My personal preference is `dynamical_family_async`.

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+
+        return response
+
+    def escaperoutes_async(self, object_name, *,
```
similarly: escape_routes_async

hhsieh00 replied:
changed

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+        """
+
+        if 'elements' in src:
+            src = OrderedDict(src['elements'])
```
Is there a reason for OrderedDict?

mkelley commented on astroquery/astorbdb/core.py:
```diff
@@ -0,0 +1,720 @@
+        except ValueError:
+            raise ValueError('Server response not readable.')
+
+        if self.query_type == 'designations':
```
`if ... elif ... elif` would be better: indicates that only one match can occur, and allows for a final `else` that can raise a `ValueError` in case of no match.

mkelley reviewed (changes requested):
Looks good and very useful in general.  Some initial comments and questions to address while you continue testing.

bsipocz commented:
Thanks @mkelley for the review. I haven't looked into it in any meaningful way yet, but I notice that there are a lot of comments on async stuff. We should remove/rework the template module as a lot has changed. E.g. there is a preference of supporting actual async/sync API and not just with the early template hackary. So I would suggest to so that, too, if the service support async, then opt into that with a keyword argument, but don't duplicate the methods with the decorator.

https://github.com/astropy/astroquery/issues/2598

or one PR that did the keyword approach: https://github.com/astropy/astroquery/pull/3201 or the ESA modules also have a `async_job` keyword to switch between sync and async query modes.

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
First draft of completed astorbdb class

First complete draft of the astorbdb class including tests and documentation

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
fixes added per comments by mkelley

removed get_raw_response lines, changed names of dynamical_family and escape_routes methods, changed OrderedDict output to regular dictionaries. Also re-ordered methods in alphabetical order.  Other minor edits.

hhsieh00 committed:
changes to documentation files

fixed dynamical_family and escape_route function names in documentation; also added astorbdb.rst to various toc files

hhsieh00 committed:
changes to documentation files

fixed dynamical_family and escape_route function names in documentation; also added astorbdb.rst to various toc files

hhsieh00 committed:
changes to documentation files

fixed dynamical_family and escape_route function names in documentation; also added astorbdb.rst to various toc files

hhsieh00 committed:
changes to documentation files

fixed dynamical_family and escape_route function names in documentation; also added astorbdb.rst to various toc files

hhsieh00 committed:
updated CHANGES.rst

hhsieh00 committed:
updated CHANGES.rst

hhsieh00 committed:
updated CHANGES.rst

hhsieh00 committed:
updated CHANGES.rst

hhsieh00 committed:
added URL checks

hhsieh00 committed:
added URL checks

hhsieh00 committed:
added URL checks

hhsieh00 committed:
added URL checks

hhsieh00 committed:
added URL checks

hhsieh00 committed:
added URL checks

hhsieh00 committed:
removed duplicate function definitions

hhsieh00 committed:
removed duplicate function definitions

hhsieh00 committed:
removed duplicate function definitions

hhsieh00 committed:
fixed code style errors

hhsieh00 committed:
fixed code style errors

hhsieh00 committed:
fixed code style errors

hhsieh00 committed:
code style fixes

hhsieh00 committed:
code style fixes

hhsieh00 committed:
code style fixes

hhsieh00 committed:
fixed mislabeled remote data doctests

hhsieh00 committed:
fixed mislabeled remote data doctests

hhsieh00 committed:
fixed mislabeled remote data doctests

mkelley reviewed (commented):
Looks good, thanks again!  I'll ask a general question for @bsipocz , this PR adds about 0.4 MB of test data.  Is that OK, or should we consider reducing it?

hhsieh00 committed:
reduced amount of test data

removed unnecessary/redundant test data

hhsieh00 committed:
reduced amount of test data

removed unnecessary/redundant test data

hhsieh00 committed:
reduced amount of test data

removed unnecessary/redundant test data

hhsieh00 committed:
fixed errors + changed quotes to single

hhsieh00 committed:
fixed errors + changed quotes to single

hhsieh00 committed:
fixed errors + changed quotes to single

hhsieh00 committed:
removed pre-commit config file

hhsieh00 committed:
removed pre-commit config file

hhsieh00 committed:
removed pre-commit config file

hhsieh00 committed:
made commenting more specific

to aid in future code maintenance

hhsieh00 committed:
made commenting more specific

to aid in future code maintenance

hhsieh00 committed:
made commenting more specific

to aid in future code maintenance

hhsieh00 committed:
removed white space

hhsieh00 committed:
removed white space

hhsieh00 committed:
removed white space

hhsieh00 committed:
renamed module from astorbdb to lowellmps

per discussions with mkelley; avoids confusion with "astorbDB" having more data than just orbital elements, and also makes the module more suitable for potential future expansion, e.g., to comet data, as additional tools are added to the Lowell Minor Planet Services

hhsieh00 committed:
renamed module from astorbdb to lowellmps

per discussions with mkelley; avoids confusion with "astorbDB" having more data than just orbital elements, and also makes the module more suitable for potential future expansion, e.g., to comet data, as additional tools are added to the Lowell Minor Planet Services

hhsieh00 committed:
renamed module from astorbdb to lowellmps

per discussions with mkelley; avoids confusion with "astorbDB" having more data than just orbital elements, and also makes the module more suitable for potential future expansion, e.g., to comet data, as additional tools are added to the Lowell Minor Planet Services

hhsieh00 committed:
Update __init__.py

hhsieh00 committed:
Updates to __init__.py files in solarsystem module

hhsieh00 committed:
Updates to __init__.py files in solarsystem module

hhsieh00 committed:
Updates to __init__.py files in solarsystem module

hhsieh00 committed:
Fixed white space error

hhsieh00 committed:
Update lowellmps.rst

hhsieh00 committed:
Updates requested by N. Moskovitz

Per solicited feedback from N. Moskovitz (PI of the Lowell astorbDB effort) pdated references in documentation page, and removed AstInfo.orbit method due to some quantities being poorly documented and/or no longer being updated

hhsieh00 committed:
Updates requested by N. Moskovitz

Per solicited feedback from N. Moskovitz (PI of the Lowell astorbDB effort) pdated references in documentation page, and removed AstInfo.orbit method due to some quantities being poorly documented and/or no longer being updated

hhsieh00 committed:
Updates requested by N. Moskovitz

Per solicited feedback from N. Moskovitz (PI of the Lowell astorbDB effort) pdated references in documentation page, and removed AstInfo.orbit method due to some quantities being poorly documented and/or no longer being updated

hhsieh00 committed:
small formatting edits to lowellmps documentation

hhsieh00 committed:
small formatting edits to lowellmps documentation

hhsieh00 committed:
trying to debug buildthedocs failure

redid astinfo_all query to update output example in case manual editing was incorrect

hhsieh00 committed:
small formatting edits

hhsieh00 committed:
small formatting edits

hhsieh00 committed:
more small formatting edits

hhsieh00 committed:
formatting

hhsieh00 committed:
Revert "formatting"

This reverts commit b31d5b700ffb243602f1a407bfa40809fb73ed99.

hhsieh00 committed:
Revert "Revert "formatting""

This reverts commit 35fecfb407052c89f3b9b5362b7301cb4aa5ff81.

hhsieh00 committed:
Revert "Revert "Revert "formatting"""

This reverts commit 4e42f469301eed18cf8c600d6406a78d9c01504a.

hhsieh00 committed:
Reapply "formatting"

This reverts commit 35fecfb407052c89f3b9b5362b7301cb4aa5ff81.

hhsieh00 committed:
Revert "Merge branch '3129-add-lowell-astorbdb-query-module' of https://github.com/hhsieh00/astroquery into 3129-add-lowell-astorbdb-query-module"

This reverts commit 3ba8d4c602db761fb0c20b4c87788755c19c06b1, reversing
changes made to b31d5b700ffb243602f1a407bfa40809fb73ed99.

hhsieh00 committed:
Revert "formatting"

This reverts commit b31d5b700ffb243602f1a407bfa40809fb73ed99.

hhsieh00 committed:
Revert "Updates requested by N. Moskovitz"

This reverts commit 200a62e5e1368c9e7f8634dcf8c547aad28ec00d.

hhsieh00 committed:
Revert "Updates to __init__.py files in solarsystem module"

This reverts commit 39d3c675b82ff9ef94b769e18ec3c28e1b0afdc4.

hhsieh00 committed:
Revert "renamed module from astorbdb to lowellmps"

This reverts commit c0fd76ca879113a907d843666da88346fb87e0be.

hhsieh00 committed:
Revert "removed white space"

This reverts commit 18b2d699124139da6fa0af4464b0d9b7168688ff.

hhsieh00 committed:
Revert "made commenting more specific"

This reverts commit 2f31596325cf6b27ee7eb1f5bfe9817c17a4c874.

hhsieh00 committed:
Revert "removed pre-commit config file"

This reverts commit 8678dfcdaed9dda1f18215b576b9ee35db971f0a.

hhsieh00 committed:
Revert "fixed errors + changed quotes to single"

This reverts commit 7fa8eb026186ce94a6616ab886f55f050ff8b8cb.

hhsieh00 committed:
Revert "reduced amount of test data"

This reverts commit 7756c521b08b979562c00653bb3a80ac49ee2548.

hhsieh00 committed:
Revert "fixed mislabeled remote data doctests"

This reverts commit 6359f96a372d41a460abdc0387d114cb805e6e3e.

hhsieh00 committed:
Revert "code style fixes"

This reverts commit ed28abae42df0d4de3545220d46ba3cbed5a1e9a.

hhsieh00 committed:
Revert "fixed code style errors"

This reverts commit 2f0326918d63921bc74734271499b22648d86756.

hhsieh00 committed:
Revert "added URL checks"

This reverts commit 84ad8b1b19d745d9c41e00dafb2659c949730d21.

hhsieh00 committed:
Revert "fixes added per comments by mkelley"

This reverts commit cb889a2600aff334bf813874854dac8aac1d9cb0.

hhsieh00 committed:
Revert "First draft of completed astorbdb class"

This reverts commit 566ac6b9dca11d75ff0146a69dbe98bad5834304.

hhsieh00 committed:
Revert "fixing more codestyle issues in core.py"

This reverts commit 5c8298a8f4cfcb1ba64cc0ed575637027a1c8d22.

hhsieh00 committed:
Revert "Fixing some codestyle issues in core.py"

This reverts commit a72f7b8d8e7f0d349ed91cf2532f70cfc73d480f.

hhsieh00 committed:
Revert "draft astroquery code for Lowell astorbdb AstInfo functionality"

This reverts commit 8920bf8a9f6dc8f970e79ebe8c70d34821c0c532.

hhsieh00 committed:
Revert "removed duplicate function definitions"

This reverts commit bf119b2f9c95a575aecf1e7109a2f51e07b2220d.

hhsieh00 committed:
Revert "added URL checks"

This reverts commit d56b2517b14af0e55687cb4b6d0eddc101a548ba.

hhsieh00 committed:
Revert "fixes added per comments by mkelley"

This reverts commit b51ff22b26e17f1d9b2772aa4c6199edf6e72008.

hhsieh00 committed:
Revert "First draft of completed astorbdb class"

This reverts commit 31581a6fc477e1f16e08bf76b87d656561187175.

hhsieh00 committed:
Revert "fixing more codestyle issues in core.py"

This reverts commit 543dea912c72f942c383791cd1a2875c47166cf3.

hhsieh00 committed:
Revert "Fixing some codestyle issues in core.py"

This reverts commit 8bb7a92adb343100ebb48fcdd5fd02fa07c03a20.

hhsieh00 committed:
Revert "draft astroquery code for Lowell astorbdb AstInfo functionality"

This reverts commit e590e4f4f2482ce9c03349b5212145f1aa44b162.

hhsieh00 committed:
Revert "updated CHANGES.rst"

This reverts commit 49700d0ef322eface4df691b654bdda06e959fda.

hhsieh00 committed:
Revert "fixes added per comments by mkelley"

This reverts commit ca91720497be72e8108f998fb7aa8b292da6067a.

hhsieh00 committed:
Revert "First draft of completed astorbdb class"

This reverts commit 5e13d19b1cf8ba4c75e8f208603f19069f8eeed8.

hhsieh00 committed:
Revert "fixing more codestyle issues in core.py"

This reverts commit 3d9d148145137f5644aa845bee28dcaee6ecbfda.

hhsieh00 committed:
Revert "Fixing some codestyle issues in core.py"

This reverts commit 70c829769c3d82d517409a95027fc49a0de84147.

hhsieh00 committed:
Revert "draft astroquery code for Lowell astorbdb AstInfo functionality"

This reverts commit f57df66c71febc28853e65b13b44851fb0192b46.

hhsieh00 committed:
Revert "changes to documentation files"

This reverts commit 4b17fb8a495e5e1cd6fb1f2c4519c77e2db3bcb8.

hhsieh00 committed:
Revert "fixes added per comments by mkelley"

This reverts commit 756d0dc2863b8125ebe51dddd6bb13bd5424f677.

hhsieh00 committed:
Revert "First draft of completed astorbdb class"

This reverts commit 646d3a253be09815f9cadb760aea8a34b567aab4.

hhsieh00 committed:
Revert "more codestyle issue fixes"

This reverts commit 804a66283d885ffb7c82ff575ffecbff5b47f350.

hhsieh00 committed:
Revert "fixing more codestyle issues in core.py"

This reverts commit 1adca45ee1576c7aea4f4e5ebade53358611da2e.

hhsieh00 committed:
Revert "Fixing some codestyle issues in core.py"

This reverts commit 84b5febe2c55e26eeba01ab583cf64d7abe2bb26.

hhsieh00 committed:
Revert "draft astroquery code for Lowell astorbdb AstInfo functionality"

This reverts commit dccfb7e7911f192bc6c78e9b2a6c0cb72d458fc0.

hhsieh00 committed:
Revert "more small formatting edits"

This reverts commit e535c6dde6cf2efa5706647b726a8ecb2a84d4f1.

hhsieh00 committed:
removed extra blank line in core.py

hhsieh00 committed:
removed automodapi code from ss/lowellmps.rst

removed automodapi code from solarsystem version of  lowellmps.rst to match other sections (e.g., jpl, imcce)

hhsieh00 committed:
fixed some errors in buildthedocs error log

mkelley reviewed (approved):
Just one comment on the documentation.

bsipocz commented on CHANGES.rst:
```diff
@@ -4,7 +4,10 @@
 New Tools and Services
 ----------------------
 
+lowellmps
```
I would consider adding an underscore in the name, that would ease readibility and we have precedent for it e.g. `esa.xmm_newton` or `ipac.nexsci.nasa_exoplanet_archive`

hhsieh00 replied:
@bsipocz I would be in favor of this.  With the nexsci name in mind in particular, what do you think about expanding it a bit more to `lowell_minor_planet_services`?  "MPS" is not really an abbreviation in common usage that I'm aware of, so spelling it out completely would be the best for readability in my opinion, at the expense of compactness.

bsipocz commented on astroquery/lowellmps/__init__.py:
```diff
@@ -0,0 +1,39 @@
+LOWELLMPS
+-------------------------
+
+:author: Henry Hsieh (hhsieh@gmail.com)
```
maybe add a 1-2 sentence description here, otherwise loose this docstring.

Also, note that we are cleaning out this author information -- the git history should hold that information much better than these files.

bsipocz commented on CHANGES.rst:
```diff
@@ -4,7 +4,10 @@
+lowellmps
+^^^^^^^^^
 
+- module added to query the Lowell Minor Planet Services database [#3203]
```
only the `AstInfo` tool was added, so be explicit about it

bsipocz commented on astroquery/lowellmps/core.py:
```diff
@@ -0,0 +1,724 @@
+
+        Returns
+        -------
+        res : A dictionary holding available AstInfo data
```
I would say drop duplicating the async/sync methods approach

bsipocz commented on astroquery/lowellmps/core.py:
```diff
@@ -0,0 +1,660 @@
+
+        response = {}
+
+        response['albedos'] = self._request('GET',
```
please reuse the existing other methods for all of these rather than recreating the requests from scratch; here and everywhere else below in this method

bsipocz commented on astroquery/lowellmps/core.py:
```diff
@@ -0,0 +1,660 @@
+            src = self._process_data_colors(src)
+
+        elif self.query_type == 'designations':
+            src = self._process_data_designations(src)
```
I would think that most these additional methods are bringing it a lot of extra scaffolding with very little benefit, so consider loosing them at least for the ones that are really just one liners.

bsipocz commented on docs/lowellmps/lowellmps.rst:
```diff
@@ -0,0 +1,250 @@
+   OrderedDict({'albedos': [{'albedo': 0.065, 'albedo_error_lower': -0.002, 'albedo_error_upper': 0.002, 'citation_bibcode': '2011PASJ...63.1117U', 'citation_url': 'https://darts.isas.jaxa.jp/astro/akari/catalogue/AcuA.html', 'count_bands_detection': 2, 'diameter': <Quantity 54.32 km>, 'diameter_error_lower': <Quantity -0.77 km>, 'diameter_error_upper': <Quantity 0.77 km>, 'eta': 0.82, 'eta_error ...
+
+Output
+======
```
there are already examples above with outputs, so I would find it helpful if this heading would be a bit more detailed

bsipocz commented on docs/lowellmps/lowellmps.rst:
```diff
@@ -0,0 +1,250 @@
+.. _astroquery.lowellmps:
+
```
If you add this, all the remote-data directives can be removed from below


```suggestion

.. as all code examples require remote-data access, thus only using this one at the very top
.. doctest-remote-data-all::

```

bsipocz reviewed (commented):
Thank you @hhsieh00 for this PR. I have a a large collection of comments, but this feels to be rather close to be merge able. The big picture items:

- I think the module could be named in a bit more descriptive way
- there are  way too many commits with a lot of revertions. Those should preferably be squashed into the commits they are reverting. 
- I have commented on the async-sync question already, there should not be a duplicate of methods any more, no usage of the decorator and if the service supports real async, then control it with a kwarg, like `async_job=True`
- I feel the API can be a bit more descriptive or better fit into the method naming approaches we use in the other modules (e.g. not just a noun but an active verb+noun, e.g. `query_albedos` -- I firmly believe this will help users in the long run)

Again, thank you so much for working on this module, I hope while the comments are numerous they won't be too complicated to be addressed.

mkelley commented on docs/lowellmps/lowellmps.rst:
```diff
@@ -0,0 +1,250 @@
+database.
+
+The development of this submodule is funded through NASA PDART
+Grant No. 80NSSC18K0987 to the `sbpy project <https://sbpy.org>`_.
```
Ah, yes, @hhsieh00 that should be 80NSSC22K0143
